File: Cardia/src/Cardia/Core/Input.cpp
#include "cdpch.hpp"
#include "Cardia/Core/Input.hpp"

#include <GLFW/glfw3.h>


namespace Cardia
{
	static GLFWwindow* GetNativeWin()
	{
		return static_cast<GLFWwindow*>(Application::Get().GetWindow().GetNativeWindow());
	}

	bool Input::IsKeyPressed(int keyCode) {
		auto state = glfwGetKey(GetNativeWin(), keyCode);
		return state == GLFW_PRESS || state == GLFW_REPEAT;
	}

	bool Input::IsMouseButtonPressed(int button) {
		return glfwGetMouseButton(GetNativeWin(), button) == GLFW_PRESS;
	}

	glm::vec2 Input::GetMousePos() {
		double x, y;
		glfwGetCursorPos(GetNativeWin(), &x, &y);

		return { static_cast<float>(x), static_cast<float>(y) };
	}

	float Input::GetMouseX() {
		return GetMousePos().x;
	}

	float Input::GetMouseY() {
		return GetMousePos().y;
	}
}

File: cardia.py/src/cardia/event/input.py
import cardia_native as _cd
from .keycode import Key, Mouse
from ..math import Vector2


class Input:
    @staticmethod
    def is_key_pressed(key: Key) -> bool:
        return _cd.is_key_pressed(int(key))

    @staticmethod
    def is_mouse_button_pressed(key: Mouse) -> bool:
        return _cd.is_mouse_button_pressed(int(key))

    @staticmethod
    def get_mouse_position() -> Vector2:
        return _cd.get_mouse_position()

    @staticmethod
    def get_mouse_x() -> float:
        return _cd.get_mouse_x()

    @staticmethod
    def get_mouse_y() -> float:
        return _cd.get_mouse_y()
